swift.go

```diff
@@ -519,8 +520,8 @@ again:
 			// Try again for a limited number of times on
 			// AuthorizationFailed or BadRequest. This allows us
 			// to try some alternate forms of the request
-			if (err == AuthorizationFailed || err == BadRequest) && retries > 0 {
```

Comment:
I think this still needs the retries logic otherwise it will retry forever won't it?

Reply:
> I think this still needs the retries logic otherwise it will retry forever won't it?

The inclusion of `shouldReattempt` in the if-condition and setting it to `false`  if the condition evaluates to `true` should prevent infinite retries, no?